docs/CUSTOM_RULES_101.md

```diff
@@ -30,54 +30,76 @@ This document is an introduction to custom rule writing for the SonarQube Java A
 
 ## Getting started
 
-The rules you are going to develop will be delivered using a dedicated, custom plugin, relying on the **SonarSource Analyzer for Java API**. In order to start working efficiently, we provide a empty template maven project, that you will fill in while following this tutorial.
+The rules you are going to develop will be delivered using a dedicated, custom plugin, relying on the **SonarSource Analyzer for Java API**. In order to start working efficiently, we provide a template maven project, that you will fill in while following this tutorial.
 
-Grab the template project from there and import it to your IDE: https://github.com/SonarSource/sonar-custom-rules-examples/tree/master/java-custom-rules
-
-This project already contains custom rules. Our goal will be to add an extra rule!
+Grab the template project by cloning this repository (https://github.com/SonarSource/sonar-java) and then importing in your IDE the sub-module: https://github.com/SonarSource/sonar-java/tree/master/docs/java-custom-rules-example
+This project already contains examples of custom rules. Our goal will be to add an extra rule!
 
 ### Looking at the POM
 
-A custom plugin is a Maven project, and before diving into code, it is important to notice a few relevant lines related to the configuration of your soon-to-be-released custom plugin.
+A custom plugin is a Maven project, and before diving into code, it is important to notice a few relevant lines related to the configuration of your soon-to-be-released custom plugin. The root of a Maven project is a file named `pom.xml`. In our case, we have two of them: `pom.xml` and `pom_self_contained.xml`. **We will only use in this tutorial the file named `pom_self_contained.xml`**, as it is completely independent from the build of the Java Analyzer, and is self contained.
 
-In the code snippet below, note the plugin API version (`<sonar.version>`) provided through the properties. It relates to the minimum version of SonarQube your plugin will support, and is generally aligned to your company's SonarQube instance. In this template, we rely on the version **8.2.0.32929** (LTS version is **7.9**, but compatibility to latest release is guaranteed when packaging the plugin).
-Note that latest released versions of the Java Analyzer are always compatible with the current LTS version of SonarQube.
+Let's start by building the custom-plugin template by using the following command:
 
-The property `<java.plugin.version>` is the minimum version of the Java Analyzer that will be required to run your custom plugin in your SonarQube instance. Consequently, as we will rely on version **6.2.0.21135** of the Java plugin, the SonarQube instance which will use the custom plugin will also need version **6.2.0.21135** of the Java Plugin as well.
+```
+mvn clean install -f pom_self_contained.xml -Dsonarqube.version=8.7.0.41497 -Dsonarjava.version=6.13.0.25138
```

Comment:
I have a concern about the versions stated here: sonarjava 6.**12**.0.24852 is included in SQ 8.7.
If I use this version instead, the compilation fails: 

```
[ERROR] /C:/GIT/sonar-java/docs/java-custom-rules-example/src/test/java/org/sonar/samples/java/checks/SpringControllerRequestMappingEntityRuleTest.java:[23,38] cannot find symbol
[ERROR]   symbol:   class FilesUtils
[ERROR]   location: package org.sonar.java.checks.verifier
[ERROR] /C:/GIT/sonar-java/docs/java-custom-rules-example/src/test/java/org/sonar/samples/java/checks/AvoidSuperClassRuleTest.java:[23,38] cannot find symbol
[ERROR]   symbol:   class FilesUtils
[ERROR]   location: package org.sonar.java.checks.verifier
[...]
```

In addition, it is not possible anymore to upgrade the plugins. It seems that we are a bit stuck here...

Reply:
This two commands should work:

```
mvn clean install -f pom_self_contained.xml -Dsonarqube.version=7.9.6 -Dsonarjava.version=6.3.2.22818
```

```
mvn clean install -f pom_self_contained.xml -Dsonarqube.version=8.7.0.41497 -Dsonarjava.version=6.12.0.24852
```